### ISSUE TITLE
Regression of #423: Stop skipping directories that start with '_'

### ISSUE THREAD

**Issue by tomwilkie** [bazelbuild/rules_go#934]
Looks like we've regressed on this one; minimal reproduction at: https://github.com/tomwilkie/gazelle-test

```bash
$ # go fetch latest vendored dependancies
$ dep ensure && dep prune 

$ # update BUILD.bazel files
$ bazel run //:gazelle  
..............
INFO: Found 1 target...
Target //:gazelle up-to-date:
  bazel-bin/gazelle_script.bash
  bazel-bin/gazelle
INFO: Elapsed time: 84.195s, Critical Path: 5.27s

INFO: Running command line: bazel-bin/gazelle

$ # try and build
$ bazel build //:test
..............
ERROR: /Users/twilkie/Documents/src/github.com/tomwilkie/gazelle-test/BUILD.bazel:13:1: no such package 'vendor/github.com/openzipkin/zipkin-go-opentracing/_thrift/gen-go/zipkincore': BUILD file not found on package path and referenced by '//:go_default_library'.
ERROR: Analysis of target '//:test' failed; build aborted.
INFO: Elapsed time: 12.356s

$ ls vendor/github.com/openzipkin/zipkin-go-opentracing/_thrift/gen-go/zipkincore
constants.go	ttypes.go
```

See #423 and #425

**Comment by jayconrod** [comment]
Digging back through history, I think I added this back in because github.com/mattn/go-sqlite3 is broken without it. There is a directory there called _example, and some of the code there breaks unless the go-sqlite3 library is built with a special tag. There's no way for Gazelle to figure that out, and that package is widely used.

I wish there were some concrete rules on what's allowed in a directory name or import path. As far as I can tell, this isn't specified. There is some special handling in the Go SDK for directories starting with '.', '_', and directories named testdata. We've tried to emulate that. go list ignores these directories. go build allows them to be imported.

In my opinion, '_' is a (somewhat ambiguous) signal that a directory should be hidden. Gazelle will continue to ignore these by default. We should provide a way to explicitly include these files though. What would work for you? Maybe an option that specifies a pattern or a list of patterns?
